tests/epyccel/modules/augassign.py

```diff
@@ -153,3 +153,9 @@ def augassign_div_2d_complex(a):
     b = a
     b /= (4.0 + 2.0j)
     return b[0][0]
+
+def augassign_func(x : float, y : float):
+    def fun1(x: 'float') -> 'float':
+        return x + 1
+    x += fun1(y)
+    return x
```

Comment:
Instead of testing a function in function which does not work in C, could we not use a simple `numpy` or `math` function? I would also add a test with arrays if possible.

Reply:
No. NumPy and math functions are translated to objects from the ast not `FunctionCall` objects so they won't illustrate this bug

Reply:
Ah, that's true. But still we should test the C printer somehow...

Reply:
I'm not sure we can for this specific problem. But in any case the problem does not exist in C as there we can directly print `x += fun1(y)`. So as long as the semantic stage doesn't mess up the tree everything is ok. The semantic stage is tested in `tests/ast`. Beyond that we can test `AugAssign` operators, but that is already done in this file

Reply:
I have added an array test: 5529f911

Reply:
If you think it is important to test an expression of the type `x += fun1(y)` in C. We could translate a module, but we either need to create a module for this in the module tests, or translate https://github.com/pyccel/pyccel/blob/devel/tests/epyccel/modules/augassign.py
but in that case this test will fail whenever any augassign test fails.

Personally I don't think it is worth it given that function calls are not a special case in C

Reply:
I think you are probably right, testing that expression in C is not worth the effort.